common/config.go

```diff
@@ -0,0 +1,240 @@
+/*
+Copyright 2019 The Kubeedge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"io/ioutil"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+type KubeEdgeCntrllerConfig struct {
+	Master              string `yaml:"master"`
+	Namespace           string `yaml:"namespace"`
+	ContentType         string `yaml:"content_type"`
+	QPS                 uint   `yaml:"qps"`
+	Burst               uint   `yaml:"burst"`
+	NodeUpdateFrequency uint   `yaml:"node_update_frequency"`
+	Kubeconfig          string `yaml:"kubeconfig"`
+}
+
+type CloudControllerSt struct {
+	Kube KubeEdgeCntrllerConfig `yaml:"kube"`
+}
+
+type CloudHubSt struct {
+	IPAddress         string `yaml:"address"`
+	Port              uint16 `yaml:"port"`
+	CA                string `yaml:"ca"`
+	Cert              string `yaml:"cert"`
+	Key               string `yaml:"key"`
+	KeepAliveInterval uint32 `yaml:"keepalive-interval"`
+	WriteTimeout      uint32 `yaml:"write-timeout"`
+	NodeLimit         uint32 `yaml:"node-limit"`
+}
+
+type DeviceControllerSt struct {
+	Kube KubeEdgeCntrllerConfig `yaml:"kube"`
+}
+
+type ControllerYaml struct {
+	Controller       CloudControllerSt  `yaml:"controller"`
+	CloudHub         CloudHubSt         `yaml:"cloudhub"`
+	DeviceController DeviceControllerSt `yaml:"devicecontroller"`
+}
+
+type ModulesSt struct {
+	Enabled []string `yaml:"enabled"`
+	//Enabled string `yaml:"enabled"`
+}
+
+type ModulesYaml struct {
+	Modules ModulesSt `yaml:"modules"`
+}
+
+type LoggingYaml struct {
+	LoggerLevel   string   `yaml:"loggerLevel,omitempty"`
+	EnableRsysLog bool     `yaml:"enableRsyslog,omitempty"`
+	LogFormatText bool     `yaml:"logFormatText,omitempty"`
+	Writers       []string `yaml:"writers,omitempty"`
+	LoggerFile    string   `yaml:"loggerFile,omitempty"`
+}
+
+func Write2File(path string, data interface{}) error {
+	d, err := yaml.Marshal(&data)
+	if err != nil {
+		return err
+	}
+
+	if err = ioutil.WriteFile(path, d, 0666); err != nil {
+		return err
+	}
+	return nil
+}
+
+func WriteControllerYamlFile(path string) error {
```

Comment:
Can't make the parameters configurable?

Reply:
The current one mentioned are defaults. Which can be made configurable, I planned to create an issue today for it so that community can contribute.